geps/gep-713/index.md

```diff
@@ -3,1155 +3,1192 @@
+
+### Status reporting
+
+#### Policy status
+
+Policy CRDs MUST define a status stanza that allows for reporting the status of the policy with respect to each scope the resource may apply.
+
+The basic status conditions are:
+
+* **Accepted**: the policy passed both syntactic validation by the API server and semantic validation enforced by the controller, such as whether the target objects exist.
+* **Enforced**: the policy's spec is guaranteed to be fully enforced, to the extent of what the controller can ensure.
+* **PartiallyEnforced**: parts of the policy's spec is guaranteed to be enforced, while other parts are known to have been superseded by other specs, to the extent of what the controller can ensure. The status should include details highlighting which parts of the policy are enforced and which parts have been superseded, with the references to all other related policies.
+* **Overridden**: the policy's spec is known to have been fully overridden by other specs. The status should include the references to the other related policies.
+
+Policy implementations SHOULD support these basic status conditions.
+
+#### Target object status
+
+Implementations of Policies kinds SHOULD put a condition into `status.Conditions` of any objects affected by the policy.
+
+That condition, if added, MUST be named according to the pattern `<metaresource-kind>Affected` (e.g. `colors.controller.k8s.io/ColorPolicyAffected`), and SHOULD include an `observedGeneration` field kept up to date when the spec of the target object changes.
+
+Implementations SHOULD use their own unique domain prefix for this condition type. Gateway API implementations, for instance, SHOULD use the same domain as in the `controllerName` field on `GatewayClass` (or some other implementation-unique domain for implementations that do not use `GatewayClass`.)
+
+E.g. – given a `Gateway` object that is targeted by a hypothetical `ColorPolicy` policy object named `policy-namespace/my-policy`, which is owned by a `colors.controller.k8s.io` controller and with status `Enforced` or `PartiallyEnforced`. The controller SHOULD add to the status of the `Gateway` object a condition `colors.controller.k8s.io/ColorPolicyAffected: true`, and reason ideally referring to the `policy-namespace/my-policy` by name.
+
+Similarly, for a hypothetical `ColorPolicy` policy that targets a specific named section of the `Gateway` object (e.g., `http-listener`), the controller SHOULD add to the status of the listener section within the `Gateway` object a condition `colors.controller.k8s.io/ColorPolicyAffected: true`.
+
+For objects that do not have a `status.Conditions` field available (`Secret` is a good example), that object SHOULD instead have an annotation of `colors.controller.k8s.io/ColorPolicyAffected: true` added instead.
```

Comment:
Future action, non-blocking: We should have a note for implementers in here to say "hey, try to batch or rate-limit your updates here or something", because if you do something like removing a Policy from a GatewayClass that has thousands of Routes rolling up to it, and you need to update all of those Routes, you run a strong risk of DOS'ing the apiserver.